cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -318,38 +320,43 @@ private void restartIfNecessary(int podId, RestartContext restartContext)
         }
 
         try {
-            RestartPlan restartPlan = restartPlan(podId, pod, restartContext);
-            if (restartPlan.needsRestart || restartPlan.needsReconfig) {
-                if (deferController(podId, restartContext)) {
-                    log.debug("{}: Pod {} is controller and there are other pods to roll", reconciliation, podId);
-                    throw new ForceableProblem("Pod " + podName(podId) + " is currently the controller and there are other pods still to roll");
-                } else {
-                    if (canRoll(podId, 60_000, TimeUnit.MILLISECONDS, false)) {
-                        // Check for rollability before trying a dynamic update so that if the dynamic update fails we can go to a full restart
-                        if (!maybeDynamicUpdateBrokerConfig(podId, restartPlan)) {
-                            log.debug("{}: Pod {} can be rolled now", reconciliation, podId);
-                            restartAndAwaitReadiness(pod, operationTimeoutMs, TimeUnit.MILLISECONDS);
+            if (isCrashlooping(pod)) {
```

Comment:
Out of interest why was it necessary to add it here? Just so that we don't spend ~7 minutes before we restart it?

Reply:
I am not saying it was necessary but without it the timeout was still 20 minutes. And it makes sense. If we want to roll pod which is crashlooping, it is probably not possible to connect to it. So checking it at the beginning of the algorithm seems to be a good idea.